### PR TITLE
task 5 GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by GNicholson95**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -47,25 +48,108 @@ describe('GET /api/topics', () => {
...
+     });
+  it('GET:200 and should respond with an articles array all article objects', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body;
+        expect(Array.isArray(articles)).toBe(true)
+        articles.forEach((article) => {
+          expect (article.hasOwnProperty('author')).toBe(true);
```
Comment (two-triangles) [inline __tests__/app.test.js]: We need to be more thorough here and and also check the type of the property values as well as checking that the keys exist!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -76,3 +160,58 @@ describe('GET/api/articles/:article_id', () => {
...
 
+describe('GET/api/articles/:article_id/comments', () => {
+  it('GET:200 and should respond with an an array of comments for the given article_id', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        expect(comments.length).toBe(11);
+        console.log(comments[0]);
```
Comment (two-triangles) [inline __tests__/app.test.js]: remove logs

File: __tests__/app.test.js [hunk 3]
```diff
@@ -76,3 +160,58 @@ describe('GET/api/articles/:article_id', () => {
...
+        console.log(comments);
+        expect(comments[1]).toMatchObject({
+          comment_id: 2,
+          votes: 14,
+          author: 'butter_bridge',
+          body: 'The beautiful thing about treasure is that it exists. Got to find out what kind of sheets these are; not cotton, not rayon, silky.',
+          article_id: 1,
+        })
+        comments.forEach((comment) => {
+          expect (comment.hasOwnProperty('comment_id')).toBe(true);
```
Comment (two-triangles) [inline __tests__/app.test.js]: same comment as above, please add in checks for the typeof the property values coming back

File: __tests__/app.test.js [hunk 4]
```diff
@@ -76,3 +160,58 @@ describe('GET/api/articles/:article_id', () => {
...
+        .then((response) => {
+          const { comments } = response.body;
+          expect(comments).toBeSortedBy('created_at',{
+            descending: true,
+          });
+        });
+    });
+  it('should throw 400 error if given a bad request', () => {
+    return request(app)
+    .get('/api/articles/banana')
```
Comment (two-triangles) [inline __tests__/app.test.js]: What endpoint is this for? 

- /api/articles/article_id
- /api/articles/article_id/comments 

are not the same and you have already tested these areas for the first endpoint of the two. You need to correct this so it represents to right endpoint.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -3,7 +3,8 @@ const request = require("supertest");
 const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
 const data = require('../db/data/test-data/index')
-const endpoints = require('../endpoints.json')
+const endpoints = require('../endpoints.json');
+const { string } = require('pg-format');
```
Comment (roisinoreilly) [inline __tests__/app.test.js:7]: was this autoimported?

File: __tests__/app.test.js [hunk 6]
```diff
@@ -47,24 +48,44 @@ describe('GET /api/topics', () => {
...
+          title: 'Living in the shadow of a great man',
+          topic: 'mitch',
+          author: 'butter_bridge',
+          body: 'I find this existence challenging',
+          created_at: '2020-07-09T20:11:00.000Z',
+          votes: 100,
+          article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+        })
+        
+        // expect(typeof article).toBe('object')
```
Comment (roisinoreilly) [inline __tests__/app.test.js:70]: delete any commented out code if you're not using it

File: __tests__/app.test.js [hunk 7]
```diff
@@ -76,3 +97,73 @@ describe('GET/api/articles/:article_id', () => {
...
+            expect(response.body.articles.length).toBe(13);
+         });
+     });
+  it('GET:200 and should respond with an articles array all article objects', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body;
+        expect(Array.isArray(articles)).toBe(true)
```
Comment (roisinoreilly) [inline __tests__/app.test.js:112]: this line is redundant once you iterate through the array on the next line

File: __tests__/app.test.js [hunk 8]
```diff
@@ -76,3 +97,73 @@ describe('GET/api/articles/:article_id', () => {
...
+        articles.forEach((article) => {
+          expect(article).toHaveProperty('author');
```
Comment (roisinoreilly) [inline __tests__/app.test.js:114]: do we need to check if all these properties exist if on the following line you're accessing them using dot notation?

File: __tests__/app.test.js [hunk 9]
```diff
@@ -76,3 +97,73 @@ describe('GET/api/articles/:article_id', () => {
...
+      });
+  });
+
+  it('should return the articles without a body property present on any of the article objects.', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body;
+        expect(Array.isArray(articles)).toBe(true)
```
Comment (roisinoreilly) [inline __tests__/app.test.js:154]: same as above comments

**Review by roisinoreilly — APPROVED** [review]
good work so far. There's still a few lines of redundant code I've left some comments on, you especially don't need to be checking that a property exists on an object and then on the next line accessing that property using dot notation, so I would take out all the .hasOwnProperty checks. It would be better to use .objectContaining or .toMatchObject, but just try using it for future tickets. There's also a line imported from pg-format that you should take out. Once you've done that feel free to merge :)

File: __tests__/app.test.js (hunk 5 above)
Comment (GNicholson95) [reply __tests__/app.test.js:7]: i think i merged a few branches code together by accident, tried deleting code from other branches but this probably slipped through

**Pull request merged**